packages/pwa-kit-extension-sdk/src/react/classes/ApplicationExtension.ts

```diff
@@ -157,14 +149,14 @@ export class ApplicationExtension<
     protected getComponentMap?(): ComponentMap
 
     /**
-     * Called on the client to deserialize the extension data that was serialized on the server.
+     * Called on the client to deserialize the extension's routes that was serialized on the server.
      *
      * @param serializedExtension - The serialized extension data.
-     * @returns DeserializedExtension - The deserialized extension data.
+     * @returns RouteProps - The deserialized extension data.
      * @throws Error if getComponentMap() is not defined.
      * @throws Error if the deserialized component cannot be found in the component map.
      */
-    private deserialize(): DeserializedExtension | null {
+    private deserializeRoutes(): RouteProps[] | null {
```

Comment:
There's a js error inside `deserializeRoutes`, on line 173: `const routes = serializedExtension.routes.map(...)`. Because the object shape for `window.__EXTENSIONS__` has changed and no longer has `routes` in it.

Reply:
Ty! Updated the shape of the `window.__EXTENSIONS__` to include the `routes` field. https://github.com/SalesforceCommerceCloud/pwa-kit/pull/2290/files#diff-152fe6a87add218e80ebf30f0f8ff4bd4242a7d406c90e1136fec332fffe6d66